### PR TITLE
Add Evtgen executable

### PR DESCRIPTION
**Pull request by clementhelsens**
BEGINRELEASENOTES
- This add a new executable to run EvtGen with the Pythia8 plugin
ENDRELEASENOTES

### CODE REVIEW

File: plugins/delphes/DelphesPythia8EvtGenReader.h [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+        allParticleOutputArrayLHEF = modularDelphes->ExportArray("allParticlesLHEF");
+        stableParticleOutputArrayLHEF = modularDelphes->ExportArray("stableParticlesLHEF");
+        partonOutputArrayLHEF = modularDelphes->ExportArray("partonsLHEF");
+      }
+    }
+
+
+    // Initialize EvtGen.
+    if (useEvtGen) {
```
Comment (tmadlener) [inline plugins/delphes/DelphesPythia8EvtGenReader.h:270]: Unnecessary (see above)

File: plugins/delphes/DelphesPythia8Reader.h [hunk 2]
```diff
@@ -24,6 +24,8 @@
 
 #include <iostream>
 
+
```
Comment (tmadlener) [inline plugins/delphes/DelphesPythia8Reader.h:27]: Can we leave this file untouched for this PR? It seems to be only whitespace changes in any case.

File: plugins/delphes/examples/EDM4Hep2LCIO_example.py [hunk 3]
```diff
@@ -45,22 +54,32 @@ def generateEvents(inputFileName, outputFileName, nEvents):
         tracks = IMPL.LCCollectionVec(EVENT.LCIO.TRACK)
         recops = IMPL.LCCollectionVec(EVENT.LCIO.RECONSTRUCTEDPARTICLE)
 
+        #print ('nreco part  ',e.ReconstructedParticles.size(),'  ntracks  ',e.EFlowTrack.size())
```
Comment (tmadlener) [inline plugins/delphes/examples/EDM4Hep2LCIO_example.py:57]: Please remove the commented print statements in this file.

File: plugins/delphes/examples/EDM4Hep2LCIO_example.py [hunk 4]
```diff
@@ -45,22 +54,32 @@ def generateEvents(inputFileName, outputFileName, nEvents):
...
+
         #loop over the reconstructed particles
         for rp in range(e.ReconstructedParticles.size()):
```
Comment (tmadlener) [inline plugins/delphes/examples/EDM4Hep2LCIO_example.py:60]: I am not entirely sure, but can't we do something like
```python
for rp in e.ReconstructedParticles:
  # do something
```

If not we should definitely add that to our TODO list

**Review by tmadlener — COMMENTED** [review]
I have a few minor requests. Could you please implement them?

Also is it OK if we squash all of these and rebase them in the end?

**Comment by vvolkl** [comment]
Since this adds a new dependency, I'd propose I migrate the whole plugin to its own repository key4hep/k4SimDelphes after this is merged (and remove it here before the next tag). Frank offered to move the ddg4 plugin to dd4hep, so we'll have a much clearer state with the dependencies then.

**Comment by tmadlener** [comment]
@vvolkl Yes, I agree. The whole plugin needs a bit of cleaning up in any case, but we can then do that in the new repository.

File: plugins/delphes/DelphesPythia8EvtGenReader.h [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+                << "output_config_file - configuration file steering the content of the edm4hep output in Tcl format,\n"
+                << "pythia_card - Pythia8 configuration file,\n"
+                << "output_file - output file in ROOT format,\n" 
+	        << "DECAY.DEC - EvtGen full decay file,\n"
+                << "evt.pdl - EvtGen particle list,\n"
+	        << "user.dec - EvtGen user decay file." << std::endl;
+      return false;
+    }
+    outputfile = argv[4];
+    if (argc==8)useEvtGen=1;
```
[the remark this answers is not in the document]
Comment (clementhelsens) [reply plugins/delphes/DelphesPythia8EvtGenReader.h:213]: done

File: plugins/delphes/DelphesPythia8EvtGenReader.h (hunk 1 above)
Comment (clementhelsens) [reply plugins/delphes/DelphesPythia8EvtGenReader.h:270]: done

File: plugins/delphes/DelphesPythia8Reader.h (hunk 2 above)
Comment (clementhelsens) [reply plugins/delphes/DelphesPythia8Reader.h:27]: done

File: plugins/delphes/examples/EDM4Hep2LCIO_example.py (hunk 3 above)
Comment (clementhelsens) [reply plugins/delphes/examples/EDM4Hep2LCIO_example.py:57]: done

File: plugins/delphes/examples/EDM4Hep2LCIO_example.py (hunk 4 above)
Comment (clementhelsens) [reply plugins/delphes/examples/EDM4Hep2LCIO_example.py:60]: yes we could in principle, I thought the index was needed for some earlier checks, I will anyway improve this script later on, thanks for reminding it to me

File: plugins/delphes/DelphesPythia8EvtGenReader.h [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+
+#include "Pythia.h"
+#include "Pythia8Plugins/CombineMatchingInput.h"
+#include "Pythia8Plugins/EvtGen.h"
+
+#include <iostream>
+
+
+
+void ConvertInput(Long64_t eventCounter,
```
[the remark this answers is not in the document]
Comment (clementhelsens) [reply plugins/delphes/DelphesPythia8EvtGenReader.h]: done

**Comment by clementhelsens** [comment]
here are all the comments implemented. I am fine for squashing the commits and rebase.

**Comment by clementhelsens** [comment]
@vvolkl , done

**Pull request merged**